docs/adrs/0078-api-versioning.md

```diff
@@ -0,0 +1,133 @@
+
+### File/Code Organization
+In the process of versioning our API, we will need to have some agreed upon patterns for organizing our files and code. There are three general areas we should consider and the file organization might differ based on the section of the code we are talking about.
+
+#### Swagger Organization
+
+**Current Setup:** In our swagger-def files we currently have a file for each api (`prime.yaml`, `ghc.yaml`, etc.). We also have some shared files in the `definitions`, `parameters`, `paths`, `responses` and `tags` directories. Some of the definitions in our swagger-def yaml files reference these shared definitions and some do not. Here is an [example](https://github.com/transcom/mymove/blob/66fdbaab15ea26e669195bf14f04a5a840d9795c/swagger-def/ghc.yaml#L3158-L3162) of where `current_address` uses the shared definitions, whereas `backup_contact` uses an internal defnition.
+
+* Option 1: Keep a singular prime swagger-def file and modify the paths to point to either v1 or v2.
+  * Pros: It would be less files to manage, especially the generated swagger and server files.
+  * Cons: It would make it difficult to tell the two APIs apart. It would introduce the risk of modifying a definition that would introduce a breaking change to one or both versions. It would make the handlers and services more confusing to keep them in order.
+* Option 2: Have an entirely separate version 2 swagger file. It would be best to also create a subdirectory under the definitions for any version 2 definitions that are not internally defined, to avoid reuse.
+  * Pros: It would be easier to keep the definitions separate and avoid accidental introduction of breaking changes. It would enable us to use the generated code for our handlers and payload/model files.
+  * Cons: More generated files to manage
+
+**Chosen Alternative:** Option 2. While it does increase the number of files, it will be clearer to differentiate and easier to maintain a separate swagger file and its accompanying generated files. [You can see an example in this commit](https://github.com/transcom/mymove/pull/10816/commits/0f387d655e098e589b10ab27b388bc771e873f09).
+
+#### Handler Organization
+
+Since we will be generating separate `*messages` and `*operations` packages for the version 2 api, it would make sense then to create a separate directory for the version 2 handlers. This will make it easier to differentiate the handlers from different versions, and ensure we are using the correct generated code for that handler.
+
+You can see an example of utilizing the generated code in the handlers and creating the necessary routing [in this commit](https://github.com/transcom/mymove/pull/10816/commits/0c4be3bab159a49f83e407830a9fecb9d27278da).
+
+#### Services Organization
+
+This is where there are a few options to organize our services. And I am currently unsure about the best way to go.
```

Comment:
**suggestion (non-blocking)**: I think this is one of the statements that can get removed now that there is a chosen option.
```suggestion
This is where there are a few options to organize our services. 
```

Reply:
Thanks for that @felipe-lee. I removed that.